src/lib.rs

```diff
@@ -162,46 +238,74 @@ pub enum RestartOptions {
 }
 
 #[derive(Clone)]
-pub struct Options {
+struct Options {
     restart: RestartOptions,
     verbose: bool,
     file_handle_flags: bool,
 }
 
-impl Options {
-    pub fn new() -> Options {
-        Options {
+pub struct Runner<C: CommandLike> {
+    commands: Vec<C>,
+    restart: RestartOptions,
+    verbose: bool,
+    file_handle_flags: bool,
+}
+
+impl<C: CommandLike + Clone> Default for Runner<C> {
+    fn default() -> Self {
+        Runner::new()
+    }
+}
+
+impl<CL: CommandLike + Clone> Runner<CL> {
```

Comment:
Same for this block: why is the `Clone` necessary here?

Reply:
Same as the other ones. I cleaned it up a bit. But the clone is required for thread boundary reasons